src/main/kotlin/html4tree/main.kt

```
@@ -123,7 +137,141 @@ internal fun read_file_identity(file: File): FileIdentity {
+            null
+        } else if (limit <= 0) {
+            ByteArray(0)
+        } else {
+            val bytes = Files.readAllBytes(path)
+            if (bytes.size <= limit) {
+                bytes
+            } else {
+                bytes.copyOf(limit)
+            }
+        }
+    } catch (e: Exception) {
```

detekt (1.23.8)

[warning] The caught exception is swallowed. The original exception could be lost.

(detekt.exceptions.SwallowedException)